src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,71 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents Deadline task with an end date.
+ */
+public class Deadline extends Task {
+    private String deadline;
+    private LocalDateTime dateline;
```

Comment:
Does `dateline` refer to the DateTime representation of the task's deadline? Personally, I think that this might be confusing as dateline has another meaning all together. I would suggest changing the names of the two attributes from deadline -> deadlineString and dateline -> deadlineDatetime!